NitroxServer/GameLogic/Unlockables/PDAStateData.cs

```diff
@@ -10,55 +10,62 @@ namespace NitroxServer.GameLogic.Unlockables
-        public ThreadSafeDictionary<NitroxTechType, PDAEntry> PartiallyUnlockedByTechType { get; set; } = new ThreadSafeDictionary<NitroxTechType, PDAEntry>();
-
+        public ThreadSafeList<NitroxTechType> AnalyzedTechTypes { get; } = new ThreadSafeList<NitroxTechType>();
+        
         /// <summary>
-        /// Gets or sets the KnownTech construct which powers the popup shown to the user when a new TechType is discovered ("New Creature Discovered!")
-        /// The KnownTech construct uses both <see cref='NitroxModel.Packets.KnownTechEntryAdd.EntryCategory.KNOWN'>KnownTech.knownTech</see> and <see cref='NitroxModel.Packets.KnownTechEntryAdd.EntryCategory.ANALYZED'>KnownTech.analyzedTech</see>
+        /// Gets or sets the log of story events present in the PDA
         /// </summary>
         [DataMember(Order = 3)]
-        public ThreadSafeList<NitroxTechType> KnownTechTypes { get; } = new ThreadSafeList<NitroxTechType>();
-
-        [DataMember(Order = 4)]
-        public ThreadSafeList<NitroxTechType> AnalyzedTechTypes { get; } = new ThreadSafeList<NitroxTechType>();
-
+        public ThreadSafeList<PDALogEntry> PdaLog { get; } = new ThreadSafeList<PDALogEntry>();
+        
         /// <summary>
         /// Gets or sets the entries that show up the the PDA's Encyclopedia
         /// </summary>
-        [DataMember(Order = 5)]
+        [DataMember(Order = 4)]
         public ThreadSafeList<string> EncyclopediaEntries { get; } = new ThreadSafeList<string>();
-
+        
         /// <summary>
-        /// Gets or sets the log of story events present in the PDA
+        /// The ids of the already scanned entities.
+        /// </summary>
+        /// <remarks>
+        /// In Subnautica, this is a Dictionary, but the value is not used, the only important thing is whether a key is stored or not.
+        /// We can therefore use it as a list.
+        /// </remarks>
+        [DataMember(Order = 5)]
+        public ThreadSafeSet<NitroxId> ScannerFragments { get; } = new();
+        
+        /// <summary>
+        /// Partially unlocked PDA entries (e.g. fragments)
         /// </summary>
         [DataMember(Order = 6)]
-        public ThreadSafeList<PDALogEntry> PdaLog { get; } = new ThreadSafeList<PDALogEntry>();
+        public ThreadSafeList<PDAEntry> ScannerPartial { get; } = new();
 
+        /// <summary>
+        /// Fully unlocked PDA entries
+        /// </summary>
         [DataMember(Order = 7)]
-        public ThreadSafeDictionary<NitroxTechType, PDAProgressEntry> CachedProgress { get; } = new ThreadSafeDictionary<NitroxTechType, PDAProgressEntry>();
+        public ThreadSafeList<NitroxTechType> ScannerComplete { get; } = new();
+
+        /// <summary>
+        /// Scanned entity's scan progress sorted by TechType.
+        /// </summary>
+        /// <remarks>
+        /// In vanilla Subnautica, scan progress is not persisted but we need to persist it to be able to send the current data to arriving players.
```

Comment:
What we want is players to have the same scan progress for the same entities.
Let's consider the following situation:
- Player A joins (no progress data)
- Player A scans a leviathan from 0% to 34%
- Player B joins (finds 34% progress data)
- Player B can start from 34% of leviathan scan progress

We need to at least keep this state in the cache while the server is running but if we don't want to persist it, that's fine as long as we keep the information on server-side to be able to send it to newly connected players.

Reply:
This seems unexpected for players and might be reported as a bug.  If I was Player B, started scanning, and saw it start at 34% then I don't think I would understand that this is picking up from where another player left off.

Reply:
It actually was an issue beforehand in the past (which we fixed), because we're replicating a "friendly" world in which players are playing together so they'll like to share progress of any kind just like scan progress on an entity. But in the future, if we want to upscale Nitrox to host servers for players that don't necessary want to share their progression, we'll be able to split this progress state to the individual "teams".

Reply:
will be removed completely